kernel/comps/virtio/src/device/network/device.rs

```diff
@@ -184,8 +204,15 @@ impl NetworkDevice {
             .send_queue
             .add_dma_buf(&[&tx_buffer], &[])
             .map_err(queue_to_network_error)?;
-        if self.send_queue.should_notify() {
-            self.send_queue.notify();
+
+        self.poll_stat.sent_packet += 1;
+
+        if self.send_queue.available_desc() == 0 {
+            // If send queue is full, we will notify as soon as possible.
```

Comment:
This is because we cannot apply the same condition to check the receive queue. The receive queue consistently contains buffers, which are either filled with packets or empty.

However, we can implement a similar check to determine if there are no empty buffers available for receiving (We have received all buffers but haven't notified device about new empty buffers). I have now incorporated this check.

Reply:
> This is because we cannot apply the same condition to check the receive queue. The receive queue consistently contains buffers, which are either filled with packets or empty.

OK, I see. It's tricky. Thanks for the explanation!